apps/www/registry/default/ui/carousel.tsx

```diff
@@ -194,6 +193,36 @@ const CarouselItem = React.forwardRef<
 })
 CarouselItem.displayName = "CarouselItem"
 
+const CarouselDots = React.forwardRef<
+  HTMLDivElement,
+  React.HTMLAttributes<HTMLDivElement>
+>(({ className, ...props }) => {
+  const { selectedIndex, scrollTo, api } = useCarousel()
+  return (
+    <div
+      className={cn(
+        "embla__dots mt-4 flex w-[calc(100vw-16px)] items-center justify-center gap-4",
+        className
+      )}
+      {...props}
+    >
+      {api?.scrollSnapList().map((_, index) => (
```

Comment:
If api is undefined, the map method might not work properly, causing an issue. 

Approaches like **{api?.scrollSnapList()?.map((_, index) => (}** 
or 
**{api?.scrollSnapList()?.length > 0 && api.scrollSnapList().map((_, index) => (}**
could be considered.

Reply:
You're right, I will apply those changes right away, thanks!